ex6.5_group_by.py

```diff
@@ -0,0 +1,13 @@
+def group_by(function, iterable):
+    dictionary = {}
+    for item in iterable:
+        if function(item) in dictionary.keys():
+            dictionary[function(item)].extend([item])
```

Comment:
Use dictionary.setdefulat  instead of this if else- 
https://www.w3schools.com/python/ref_dictionary_setdefault.asp